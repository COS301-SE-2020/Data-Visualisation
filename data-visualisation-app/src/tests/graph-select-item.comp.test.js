import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import GraphSelectItem from '../graph-selection/graph-select-item.comp';

test('renders List Item representing a Graph-Type, and that sets the selected graph type once clicked', () => {
  let selectedTypeMock = '';
  const typeMock = {
    type: 'bar-chart',
    name: 'Bar Chart',
    image: './graph-images/bar-chart-1.png',
  };

  const setGraphMock = (type) => {
    selectedTypeMock = type;
  };

  const comp = <GraphSelectItem graphType={typeMock} setGraph={setGraphMock} />;
  const { queryByAltText, getByAltText } = render(comp);

  let tree = renderer.create(comp).toJSON();
  expect(tree).toMatchSnapshot();

  expect(queryByAltText(new RegExp(typeMock.type))).toBeTruthy();
  //manually trigger onClick
  fireEvent.click(getByAltText(new RegExp(typeMock.type)));

  expect(selectedTypeMock).toBe(typeMock.type);
});
