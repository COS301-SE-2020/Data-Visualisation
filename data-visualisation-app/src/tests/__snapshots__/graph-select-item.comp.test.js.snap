// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List Item representing a Graph-Type, and that sets the selected graph type once clicked 1`] = `
<div
  className="Graph-Select-Item"
>
  <div
    className="title"
  >
    Bar Chart
  </div>
  <img
    alt="bar-chart"
    className="img"
    onClick={[Function]}
    src="./graph-images/bar-chart-1.png"
  />
</div>
`;
